sql/distsql: TestAggregatorAgainstProcessor failed

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3471661&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3471661&tab=artifacts#/) on master @ [bfdcfa00836232b910cb0fd109ebedf8913b1466](https://github.com/cockroachdb/cockroach/commits/bfdcfa00836232b910cb0fd109ebedf8913b1466):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor216664919
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor216664919
    panic.go:965: expected *DGeometry, found *tree.DArray
        goroutine 11 [running]:
        runtime/debug.Stack(0xc0005102f0, 0x4045940, 0xc000b711d0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4045940, 0xc000b711d0)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x586bfa8, 0xc000ea9b30, 0xc000ea9b30)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3026 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stExtentAgg).Add(0xc0006bb548, 0x58512e8, 0xc000072108, 0x59817c8, 0xc000ea9b30, 0x0, 0x0, 0x0, 0xc0005105e0, 0x1e1d693)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1003 +0x76
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000345900, 0xc0002096c0, 0x9, 0x9, 0xc000f4f350, 0x2, 0x8, 0xc000f13bd0, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc000345900, 0xc0002096c0, 0x9, 0x9, 0x1, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc000345900, 0x0, 0x0, 0x58512e8, 0xc0005108a0, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc000345900, 0x4a8e2f0, 0xc0007ba2c0, 0xc000f9ccf0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc00054d080, 0x1, 0x0, 0x0, 0x0, 0xc000f2de00, 0x1, 0x1, 0xc000f2de18, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc00054d080)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc00054d080, 0x4a8e410)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (2.11s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472082&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472082&tab=artifacts#/) on master @ [bfdcfa00836232b910cb0fd109ebedf8913b1466](https://github.com/cockroachdb/cockroach/commits/bfdcfa00836232b910cb0fd109ebedf8913b1466):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor298684345
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor298684345
    panic.go:965: expected *DGeometry, found *tree.DArray
        goroutine 51 [running]:
        runtime/debug.Stack(0xc000e34060, 0x4045940, 0xc00063e730)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4045940, 0xc00063e730)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x586bfa8, 0xc000c8b080, 0xc000c8b080)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3026 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stCollectAgg).Add(0xc0007977d0, 0x58512e8, 0xc00019e010, 0x59817c8, 0xc000c8b080, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:875 +0xf7
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000379900, 0xc000a7eb00, 0x7, 0x7, 0xc0006fdbf0, 0x1, 0x8, 0xc0013273b0, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc000379900, 0xc000a7eb00, 0x7, 0x7, 0x7, 0x7)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc000379900, 0x0, 0x0, 0x58512e8, 0xc000e348a0, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc000379900, 0x4a8e2f0, 0xc000a9cf20, 0xc00067f9e0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc0002eed80, 0x1, 0x0, 0x0, 0x0, 0xc00104d200, 0x1, 0x1, 0xc00104d218, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc0002eed80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc0002eed80, 0x4a8e410)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (3.96s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472040&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472040&tab=artifacts#/) on master @ [fa619366b86a8b487c3a6ffed4fab1d818092ea4](https://github.com/cockroachdb/cockroach/commits/fa619366b86a8b487c3a6ffed4fab1d818092ea4):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor593001024
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor593001024
    panic.go:965: expected *DGeometry, found *tree.DArray
        goroutine 55 [running]:
        runtime/debug.Stack(0xc0011b2290, 0x4055c20, 0xc000704ae0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4055c20, 0xc000704ae0)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x58d0448, 0xc000fd6750, 0xc000fd6750)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3026 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc0007adec0, 0x58b5788, 0xc000072108, 0x59e6428, 0xc000fd6750, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:728 +0x8f
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000f00000, 0xc0001a2480, 0xb, 0xb, 0xc000da7e50, 0x3, 0x8, 0xc0012225a0, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc000f00000, 0xc0001a2480, 0xb, 0xb, 0xb, 0xb)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc000f00000, 0x0, 0x0, 0x58b5788, 0xc0011b28a0, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc000f00000, 0x4a9e9b8, 0xc0004af340, 0xc000dd0ab0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc00038c780, 0x1, 0x0, 0x0, 0x0, 0xc000cec0d8, 0x1, 0x1, 0xc000cec0f0, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc00038c780)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc00038c780, 0x4a9ead8)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (7.81s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472288&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472288&tab=artifacts#/) on master @ [fa619366b86a8b487c3a6ffed4fab1d818092ea4](https://github.com/cockroachdb/cockroach/commits/fa619366b86a8b487c3a6ffed4fab1d818092ea4):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor933554743
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor933554743
    panic.go:965: expected *DBytes, found *tree.DArray
        goroutine 30 [running]:
        runtime/debug.Stack(0xc0005c82d8, 0x4055c20, 0xc000bb1f80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4055c20, 0xc000bb1f80)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDBytes(0x58d0448, 0xc00104c420, 0xc00104c420, 0x58d0448)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:1371 +0xf0
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*concatAggregate).Add(0xc000291260, 0x58b5788, 0xc000072108, 0x59e6428, 0xc00104c420, 0xc000bb1f60, 0x1, 0x1, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1487 +0x14c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc001207200, 0xc000848d20, 0xa, 0xa, 0xc0014e1060, 0x1, 0x8, 0xc000449680, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc001207200, 0xc000848d20, 0xa, 0xa, 0xa, 0xa)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc001207200, 0x0, 0x0, 0x58b5788, 0xc0005c88a0, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc001207200, 0x4a9e9b8, 0xc0017ee840, 0xc0005d70e0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc000c1e480, 0x1, 0x0, 0x0, 0x0, 0xc00050bfe0, 0x1, 0x1, 0xc0009b2000, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000c1e480)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc000c1e480, 0x4a9ead8)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (0.73s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3471650&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3471650&tab=artifacts#/) on master @ [bfdcfa00836232b910cb0fd109ebedf8913b1466](https://github.com/cockroachdb/cockroach/commits/bfdcfa00836232b910cb0fd109ebedf8913b1466):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor288600839
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:971: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor288600839
    panic.go:971: expected *DGeometry, found *tree.DArray
        goroutine 7 [running]:
        runtime/debug.Stack(0xc0dd240, 0x1, 0xc0007f0a80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0xca
        panic(0x6005080, 0xc0007f0a80)
        	/usr/local/go/src/runtime/panic.go:971 +0x499
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x9389a08, 0xc0005939b0, 0xc0005939b0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3026 +0x145
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc000d4c8c0, 0x936ed48, 0xc0001a6010, 0x94a03f8, 0xc0005939b0, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:728 +0xaf
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0001c7200, 0xc0011c3e00, 0x8, 0x8, 0x0, 0x0, 0x0, 0xc000fe2c80, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x58e
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0001c7200, 0xc0011c3e00, 0x8, 0x8, 0x8, 0x8)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0x145
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0001c7200, 0x0, 0x0, 0xc000e80960, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:506 +0x185
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0001c7200, 0x6a55618, 0xc0003acf20, 0xc000101400, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:710 +0x2ae
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc000102c00, 0x0, 0x0, 0x0, 0x0, 0xc00039af78, 0x1, 0x1, 0xc00039af90, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0x17bc
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000102c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1c38
        testing.tRunner(0xc000102c00, 0x6a55738)
        	/usr/local/go/src/testing/testing.go:1193 +0x203
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x5d8
--- FAIL: TestAggregatorAgainstProcessor (1.24s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472412&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472412&tab=artifacts#/) on master @ [af8d186483d460e244073fe49fade1d7d1467d45](https://github.com/cockroachdb/cockroach/commits/af8d186483d460e244073fe49fade1d7d1467d45):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor710156156
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor710156156
    panic.go:965: expected *DGeometry, found *tree.DArray
        goroutine 52 [running]:
        runtime/debug.Stack(0xc00148e060, 0x4055c20, 0xc000b0a030)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4055c20, 0xc000b0a030)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x58d0448, 0xc001649a40, 0xc001649a40)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3026 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stCollectAgg).Add(0xc0013c24e0, 0x58b5788, 0xc00019e010, 0x59e6428, 0xc001649a40, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:875 +0xf7
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0001b6580, 0xc00164da40, 0xa, 0xa, 0xc0013de090, 0x3, 0x8, 0xc0013dc320, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc0001b6580, 0xc00164da40, 0xa, 0xa, 0xa, 0xa)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc0001b6580, 0x2, 0x1, 0xc00148e8a0, 0x48dba7, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc0001b6580, 0x4a9e9b8, 0xc0005ae000, 0xc001784bd0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc0006cca80, 0x1, 0x0, 0x0, 0x0, 0xc001638018, 0x1, 0x1, 0xc001638030, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc0006cca80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc0006cca80, 0x4a9ead8)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (5.29s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472647&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472647&tab=artifacts#/) on master @ [af8d186483d460e244073fe49fade1d7d1467d45](https://github.com/cockroachdb/cockroach/commits/af8d186483d460e244073fe49fade1d7d1467d45):


```
=== RUN   TestAggregatorAgainstProcessor
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor100246775
    test_log_scope.go:80: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAggregatorAgainstProcessor100246775
    panic.go:965: expected *DString, found *tree.DArray
        goroutine 116 [running]:
        runtime/debug.Stack(0xc0011c02d8, 0x4055c20, 0xc000bac830)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x4055c20, 0xc000bac830)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDString(0x58d0448, 0xc000c19cb0, 0xc000c19cb0, 0x58d0448)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:1148 +0xe6
        github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*concatAggregate).Add(0xc00177ac00, 0x58b5788, 0xc000124010, 0x59e6428, 0xc000c19cb0, 0xc000bac810, 0x1, 0x1, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1489 +0x228
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc00013c580, 0xc00030c6c0, 0xc, 0xc, 0xc0012489f0, 0x2, 0x8, 0xc00052aff0, 0x5, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:792 +0x39c
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc00013c580, 0xc00030c6c0, 0xc, 0xc, 0xc, 0xc)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1ce
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc00013c580, 0x0, 0x0, 0x58b5788, 0xc0011c08a0, 0x451449)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:437 +0x22b
        github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc00013c580, 0x4a9e9b8, 0xc000508dc0, 0xc0000eaf30, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:688 +0x14e
        github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0xc000580780, 0x1, 0x0, 0x0, 0x0, 0xc00158be90, 0x1, 0x1, 0xc00158bea8, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:191 +0xfff
        github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000580780)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:312 +0x1485
        testing.tRunner(0xc000580780, 0x4a9ead8)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestAggregatorAgainstProcessor (9.18s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/distsql/columnar_operators_test.go
+++ b/pkg/sql/distsql/columnar_operators_test.go
@@ -205,7 +205,8 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 									aggFnInputTypes[j] = randgen.RandType(rng)
 								}
 								// There is a special case for some functions when at
-								// least one argument is a tuple.
+								// least one argument is a tuple or an array of
+								// tuples.
 								// Such cases pass GetAggregateInfo check below,
 								// but they are actually invalid, and during normal
 								// execution it is caught during type-checking.
@@ -220,7 +221,7 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 									execinfrapb.StUnion,
 									execinfrapb.StCollect:
 									for _, typ := range aggFnInputTypes {
-										if typ.Family() == types.TupleFamily {
+										if typ.Family() == types.TupleFamily || (typ.Family() == types.ArrayFamily && typ.ArrayContents().Family() == types.TupleFamily) {
 											invalid = true
 											break
 										}

